README.rst
========
Overview
========

.. start-badges

.. list-table::
    :stub-columns: 1

    * - docs
      - |docs|
    * - tests
      - | |travis| |appveyor| |requires|
        | |codecov|
    * - package
      - | |version| |wheel| |supported-versions| |supported-implementations|
        | |commits-since|
.. |docs| image:: https://readthedocs.org/projects/python-publishextrator/badge/?style=flat
    :target: https://python-publishextrator.readthedocs.io/
    :alt: Documentation Status

.. |travis| image:: https://api.travis-ci.com/JonasBrother97/python-publishextrator.svg?branch=master
    :alt: Travis-CI Build Status
    :target: https://travis-ci.com/github/JonasBrother97/python-publishextrator

.. |appveyor| image:: https://ci.appveyor.com/api/projects/status/github/JonasBrother97/python-publishextrator?branch=master&svg=true
    :alt: AppVeyor Build Status
    :target: https://ci.appveyor.com/project/JonasBrother97/python-publishextrator

.. |requires| image:: https://requires.io/github/JonasBrother97/python-publishextrator/requirements.svg?branch=master
    :alt: Requirements Status
    :target: https://requires.io/github/JonasBrother97/python-publishextrator/requirements/?branch=master

.. |codecov| image:: https://codecov.io/gh/JonasBrother97/python-publishextrator/branch/master/graphs/badge.svg?branch=master
    :alt: Coverage Status
    :target: https://codecov.io/github/JonasBrother97/python-publishextrator

.. |version| image:: https://img.shields.io/pypi/v/publishextrator.svg
    :alt: PyPI Package latest release
    :target: https://pypi.org/project/publishextrator

.. |wheel| image:: https://img.shields.io/pypi/wheel/publishextrator.svg
    :alt: PyPI Wheel
    :target: https://pypi.org/project/publishextrator

.. |supported-versions| image:: https://img.shields.io/pypi/pyversions/publishextrator.svg
    :alt: Supported versions
    :target: https://pypi.org/project/publishextrator

.. |supported-implementations| image:: https://img.shields.io/pypi/implementation/publishextrator.svg
    :alt: Supported implementations
    :target: https://pypi.org/project/publishextrator

.. |commits-since| image:: https://img.shields.io/github/commits-since/JonasBrother97/python-publishextrator/v0.0.4.svg
    :alt: Commits since latest release
    :target: https://github.com/JonasBrother97/python-publishextrator/compare/v0.0.0...master



.. end-badges

PublishNews extractor

* Free software: BSD 2-Clause License

Installation
============

::

    pip install publishextrator

You can also install the in-development version with::

    pip install https://github.com/JonasBrother97/python-publishextrator/archive/master.zip


Documentation
=============


https://python-publishextrator.readthedocs.io/


Development
===========

To run all the tests run::

    tox

Note, to combine the coverage data from all the tox environments run:

.. list-table::
    :widths: 10 90
    :stub-columns: 1

    - - Windows
      - ::

            set PYTEST_ADDOPTS=--cov-append
            tox

    - - Other
      - ::

            PYTEST_ADDOPTS=--cov-append tox


CHANGELOG.rst

Changelog
=========

0.0.4 (2021-11-25)
------------------

* First release on PyPI.


tests/test_publishextrator.py

from publishextrator.cli import main


def test_main():
    assert main([]) == 0


src/publishextrator/cli.py
"""
Module that contains the command line app.

Why does this file exist, and why not put this in __main__?

  You might be tempted to import things from __main__ later, but that will cause
  problems: the code will get executed twice:

  - When you run `python -mpublishextrator` python will execute
    ``__main__.py`` as a script. That means there won't be any
    ``publishextrator.__main__`` in ``sys.modules``.
  - When you import __main__ it will get executed again (as a module) because
    there's no ``publishextrator.__main__`` in ``sys.modules``.

  Also see (1) from http://click.pocoo.org/5/setuptools/#setuptools-integration
"""
import sys


def main(argv=sys.argv):
    """
    Args:
        argv (list): List of arguments

    Returns:
        int: A return code

    Does stuff.
    """
    print(argv)
    return 0


src/publishextrator/file1.py
import requests
from bs4 import BeautifulSoup

class Periodo():
    def __init__(self, ano):
        self.ano = ano
    def extrair(self):
        global tabela
        resp = requests.get(url)
        soup = BeautifulSoup(resp.text, 'html.parser')
        titulos = soup.find_all("div", class_="pn-ranking-livro-nome")
        titulos_separado = [titulo.text.strip() for titulo in titulos]
        autores = soup.find_all("div", class_="pn-ranking-livro-autor")
        autores_separado = [autor.text.strip() for autor in autores]
        editoras = soup.find_all("div", class_="pn-ranking-livro-editora")
        editoras_separado = [editora.text.strip() for editora in editoras]
        resumos = soup.find_all("div", class_="pn-ranking-livro-resumo")
        resumos_separado = [resumo.text.strip() for resumo in resumos]
        isbns = soup.find_all("div", class_="pn-ranking-livro-isbn")
        isbns_separado = [isbn.text.strip() for isbn in isbns]
        categorias = soup.find_all("div", class_="pn-ranking-livro-categoria")
        categorias_separado = [categoria.text.strip() for categoria in categorias]
        paginas = soup.find_all("div", class_="pn-ranking-livro-paginas")
        paginas_separado = [pagina.text.strip() for pagina in paginas]
        tabela = {
            'titulo': titulos_separado,
            'autor': autores_separado,
            'editora': editoras_separado,
        #    'resumo': resumos_separado,
            'isbn': isbns_separado,
            'categoria': categorias_separado,
            'paginas': paginas_separado
        }
    def tabela(self):
        return tabela

class Semanal(Periodo):
    def __init__(self, ano, mes, dia):
        super().__init__(ano)
        self.mes = mes
        self.dia = dia
    def puxar(self):
        global url
        url = "https://www.publishnews.com.br/ranking/semanal/0/2020/12/18/0/0"
        a = list(url)
        a[49:53] = str(self.ano)
        a[54:56] = str(self.mes)
        a[57:59] = str(self.dia)
        url = "".join(a)

class Mensal(Periodo):
    def __init__(self, ano, mes):
        super().__init__(ano)
        self.mes = mes
    def puxar(self):
        global url
        url = "https://www.publishnews.com.br/ranking/mensal/0/2020/12/0/0"
        a = list(url)
        a[48:52] = str(self.ano)
        a[53:55] = str(self.mes)
        url = "".join(a)

class Anual(Periodo):
    def __init__(self, ano):
        super().__init__(ano)
    def puxar(self):
        global url
        url = "https://www.publishnews.com.br/ranking/anual/0/2020/0/0"
        a = list(url)
        a[47:51] = str(self.ano)
        url = "".join(a)


src/publishextrator/__main__.py
"""
Entrypoint module, in case you use `python -mpublishextrator`.


Why does this file exist, and why __main__? For more info, read:

- https://www.python.org/dev/peps/pep-0338/
- https://docs.python.org/2/using/cmdline.html#cmdoption-m
- https://docs.python.org/3/using/cmdline.html#cmdoption-m
"""
import sys

from publishextrator.cli import main

if __name__ == "__main__":

    class Periodo():
        def __init__(self, ano):
            self.ano = ano
        def extrair(self):
            global tabela
            resp = requests.get(url)
            soup = BeautifulSoup(resp.text, 'html.parser')
            titulos = soup.find_all("div", class_="pn-ranking-livro-nome")
            titulos_separado = [titulo.text.strip() for titulo in titulos]
            autores = soup.find_all("div", class_="pn-ranking-livro-autor")
            autores_separado = [autor.text.strip() for autor in autores]
            editoras = soup.find_all("div", class_="pn-ranking-livro-editora")
            editoras_separado = [editora.text.strip() for editora in editoras]
            resumos = soup.find_all("div", class_="pn-ranking-livro-resumo")
            resumos_separado = [resumo.text.strip() for resumo in resumos]
            isbns = soup.find_all("div", class_="pn-ranking-livro-isbn")
            isbns_separado = [isbn.text.strip() for isbn in isbns]
            categorias = soup.find_all("div", class_="pn-ranking-livro-categoria")
            categorias_separado = [categoria.text.strip() for categoria in categorias]
            paginas = soup.find_all("div", class_="pn-ranking-livro-paginas")
            paginas_separado = [pagina.text.strip() for pagina in paginas]
            tabela = {
                'titulo': titulos_separado,
                'autor': autores_separado,
                'editora': editoras_separado,
            #    'resumo': resumos_separado,
                'isbn': isbns_separado,
                'categoria': categorias_separado,
                'paginas': paginas_separado
            }
        def tabela(self):
            return tabela

    class Semanal(Periodo):
        def __init__(self, ano, mes, dia):
            super().__init__(ano)
            self.mes = mes
            self.dia = dia
        def puxar(self):
            global url
            url = "https://www.publishnews.com.br/ranking/semanal/0/2020/12/18/0/0"
            a = list(url)
            a[49:53] = str(self.ano)
            a[54:56] = str(self.mes)
            a[57:59] = str(self.dia)
            url = "".join(a)

    class Mensal(Periodo):
        def __init__(self, ano, mes):
            super().__init__(ano)
            self.mes = mes
        def puxar(self):
            global url
            url = "https://www.publishnews.com.br/ranking/mensal/0/2020/12/0/0"
            a = list(url)
            a[48:52] = str(self.ano)
            a[53:55] = str(self.mes)
            url = "".join(a)

    class Anual(Periodo):
        def __init__(self, ano):
            super().__init__(ano)
        def puxar(self):
            global url
            url = "https://www.publishnews.com.br/ranking/anual/0/2020/0/0"
            a = list(url)
            a[47:51] = str(self.ano)
            url = "".join(a)

    sys.exit(main())
